packages/manager/apps/web/client/app/domain/zone/activate/activate.routing.js

```diff
@@ -1,27 +1,45 @@
+import get from 'lodash/get';
+
+const commonResolves = {
+  autoPayWithPreferredPaymentMethod: (ovhPaymentMethod) =>
+    ovhPaymentMethod.hasDefaultPaymentMethod(),
+
+  serviceName: /* @ngInject */ ($transition$) =>
+    $transition$.params().productId,
+
+  serviceOption: /* @ngInject */ (
+    $translate,
+    Alerter,
+    serviceName,
+    DomainDnsZoneActivateService,
+  ) =>
+    DomainDnsZoneActivateService.getServiceOption(
+      serviceName,
+    ).catch((error) => Alerter.error(
+      $translate.instant('web_domain_zone_activation_error', {
+        message: get(error, 'data.message', error),
+      }),
+      'app.domain.product.zoneactivate',
+    )),
+
+  goBack: /* @ngInject */ (goToZone) => goToZone,
+};
+
 export default /* @ngInject */ ($stateProvider) => {
   $stateProvider.state('app.domain.product.zone.activate', {
     url: '/activate',
-    views: {
-      modal: {
-        component: 'domainZoneActivate',
-      },
-    },
-    layout: 'modal',
+    component: 'domainZoneActivate',
     resolve: {
-      goBack: /* @ngInject */ (goToZone) => goToZone,
+      ...commonResolves,
     },
   });
 
   $stateProvider.state('app.domain.alldom.zone.activate', {
     url: '/activate',
-    views: {
-      modal: {
-        component: 'domainZoneActivate',
-      },
-    },
-    layout: 'modal',
+    component: 'domainZoneActivate',
     resolve: {
-      goBack: /* @ngInject */ (goToZone) => goToZone,
+      ...commonResolves,
```

Comment:
Nitpicking: we could simply 
```suggestion
      resolve: commonResolves,
```
for now :)